cli/command/container/attach.go

```diff
@@ -64,36 +65,41 @@ func NewAttachCommand(dockerCli command.Cli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.BoolVar(&opts.noStdin, "no-stdin", false, "Do not attach STDIN")
-	flags.BoolVar(&opts.proxy, "sig-proxy", true, "Proxy all received signals to the process")
-	flags.StringVar(&opts.detachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
+	flags.BoolVar(&opts.NoStdin, "no-stdin", false, "Do not attach STDIN")
+	flags.BoolVar(&opts.Proxy, "sig-proxy", true, "Proxy all received signals to the process")
+	flags.StringVar(&opts.DetachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
 	return cmd
 }
 
-func runAttach(dockerCli command.Cli, opts *attachOptions) error {
-	ctx := context.Background()
+// RunAttach executes an `attach` command
+func RunAttach(dockerCli command.Cli, opts *AttachOptions) error {
+	return RunAttachWithContext(context.Background(), dockerCli, opts)
+}
+
+// RunAttachWithContext executes an `attach` command
+func RunAttachWithContext(ctx context.Context, dockerCli command.Cli, opts *AttachOptions) error {
```

Comment:
Given that `runAttach` was un-exported before this, I think it's fine to only have one function (`RunAttach`) and have that accept a context; we can probably create and pass the context in `NewAttachCommand` / `RunE`

Reply:
right, just wanted to be homogeneous with `RunExecWithContext` in method naming

Reply:
Looking at `RunExec`, I see it was exported for compose as well;

- https://github.com/docker/cli/pull/3436

These functions were never intended to be exported "as-is", and I'd be ok with changing the signature of `RunExec` to accept a context (we want to add contexts to these as well for OTEL integration; see https://github.com/docker/cli/pull/4556